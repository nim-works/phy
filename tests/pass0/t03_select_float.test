discard """
  output: "(Done 2)"
"""
(TypeDefs
  (ProcTy (Int 8))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) 0 (Locals) (List
    (Block (Params)
      (Select (Float 8) (FloatVal 20.0)
        (Choice (FloatVal 19.0) (Goto 1))
        (Choice (FloatVal 20.0) (Goto 2)))
    )
    (Block (Params)
      (Return (IntVal 1))
    )
    (Block (Params)
      (Return (IntVal 2))
    )
  ))
)