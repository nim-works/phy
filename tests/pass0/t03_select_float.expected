.type t0 (Int)
.type t1 (Float)
.type t2 (Proc t0)
.start t2 p0
  LdImmFloat 20.0
  LdImmFloat 19.0
  EqFloat
  Branch L5 1
  Jmp L7
.label L5
  LdImmInt 1
  Ret
.label L7
  LdImmInt 2
  Ret
.end
